File: python/year2015day02.py
#!/usr/bin/env python

def run(input):
    paper = 0
    ribbon = 0
    lines = input.splitlines()
    for line in lines:
        parts = line.split('x')
        w = int(parts[0])
        h = int(parts[1])
        d = int(parts[2])
        paper += getRequiredPaper(w, h, d)
        ribbon += getRequiredRibbon(w, h, d)
    print(paper)
    print(ribbon)


def getRequiredPaper(w, h, d):
    a = w * h
    b = h * d
    c = w * d
    areas = [a, b, c]
    areas.sort()
    smallest = areas[0]
    return (a + b + c) * 2 + smallest


def getRequiredRibbon(w, h, d):
    sides = [w, h, d]
    sides.sort()
    bow = w * h * d
    return (sides[0] + sides[1]) * 2 + bow

File: csharp/Year2015Day02.cs
namespace Aoc;

public static class Year2015Day02
{
    public static void Run(string input)
    {
        var paper = 0;
        var ribbon = 0;
        var lines = input.Split("\n").Select(o => o.Trim());
        foreach (var line in lines)
        {
            var parts = line.Split('x');
            var w = int.Parse(parts[0]);
            var h = int.Parse(parts[1]);
            var d = int.Parse(parts[2]);
            paper += GetRequiredPaper(w, h, d);
            ribbon += GetRequiredRibbon(w, h, d);
        }
        Console.WriteLine(paper);
        Console.WriteLine(ribbon);
    }

    private static int GetRequiredPaper(int w, int h, int d)
    {
        var a = w * h;
        var b = h * d;
        var c = w * d;
        var areas = new List<int> { a, b, c };
        areas.Sort();
        var smallest = areas.First();
        return (a + b + c) * 2 + smallest;
    }

    private static int GetRequiredRibbon(int w, int h, int d)
    {
        var sides = new List<int> { w, h, d };
        sides.Sort();
        var bow = w * h * d;
        return (sides[0] + sides[1]) * 2 + bow;
    }
}